src/OrleansAWSUtils/Statistics/DynamoDBStatisticsPublisher.cs

```diff
@@ -0,0 +1,138 @@
+        private const string ROW_KEY_PROPERTY_NAME = "RowKey";
+        private const string DEPLOYMENT_ID_PROPERTY_NAME = "DeploymentId";
+        private const string TIME_PROPERTY_NAME = "Time";
+        private const string ADDRESS_PROPERTY_NAME = "Address";
+        private const string NAME_PROPERTY_NAME = "Name";
+        private const string HOSTNAME_PROPERTY_NAME = "HostName";
+        private const string STATISTIC_PROPERTY_NAME = "Statistic";
+        private const string STATVALUE_PROPERTY_NAME = "StatValue";
+        private const string ISDELTA_PROPERTY_NAME = "IsDelta";
+        private const string DATE_TIME_FORMAT = "yyyy-MM-dd-" + "HH:mm:ss.fff 'GMT'";
+
+        private string deploymentId;
+        private string address;
+        private string name;
+        private bool isSilo;
+        private long clientEpoch;
+        private int counter;
+        private string hostName;
+        private string tableName;
+
+        private DynamoDBStorage storage;
+        private readonly Logger logger;
+
+        public DynamoDBStatisticsPublisher()
+        {
+            logger = LogManager.GetLogger(this.GetType().Name, LoggerType.Runtime);
+        }
+
+        public Task Init(bool isSilo, string storageConnectionString, string deploymentId, string address, string siloName, string hostName)
+        {
+            this.deploymentId = deploymentId;
+            this.address = address;
+            name = isSilo ? siloName : hostName;
+            this.hostName = hostName;
+            this.isSilo = isSilo;
+            if (!this.isSilo)
+            {
+                clientEpoch = SiloAddress.AllocateNewGeneration();
+            }
+            counter = 0;
+            tableName = isSilo ? "OrleansSiloStatistics" : "OrleansClientStatistics";
+
+            storage = new DynamoDBStorage(storageConnectionString, logger);
+
+            return storage.InitializeTable(tableName,
+               new List<KeySchemaElement>
+               {
+                    new KeySchemaElement { AttributeName = PARTITION_KEY_PROPERTY_NAME, KeyType = KeyType.HASH },
+                    new KeySchemaElement { AttributeName = ROW_KEY_PROPERTY_NAME, KeyType = KeyType.RANGE }
+               },
+               new List<AttributeDefinition>
+               {
+                    new AttributeDefinition { AttributeName = PARTITION_KEY_PROPERTY_NAME, AttributeType = ScalarAttributeType.S },
+                    new AttributeDefinition { AttributeName = ROW_KEY_PROPERTY_NAME, AttributeType = ScalarAttributeType.S }
+               });
+        }
+
+        public Task ReportStats(List<ICounter> statsCounters)
+        {
+            try
+            {
+                var toWrite = new List<Dictionary<string, AttributeValue>>();
+                foreach (var counter in statsCounters)
+                {
+                    var fields = ParseCounter(counter);
+                    if (fields == null)
```

Comment:
If this condition were inverted and `toWrite.Add(fields)` put in, you wouldn't need the `continue`.